packager/packager.cc

```diff
@@ -1259,6 +1283,23 @@ struct PackageSpecBodyWalk {
     }
 
 private:
+    string buildValidLayersStr(const core::GlobalState &gs) {
+        auto &validLayers = gs.packageDB().layers();
+        ENFORCE(validLayers.size() > 0);
+        if (validLayers.size() == 1) {
+            return string(validLayers.front().shortName(gs));
+        }
+        string result = "";
+        for (int i = 0; i < validLayers.size() - 1; i++) {
+            if (validLayers.size() > 2) {
+                result += core::ErrorColors::format("`{}`, ", validLayers[i].shortName(gs));
+            } else {
+                result += core::ErrorColors::format("`{}` ", validLayers[i].shortName(gs));
+            }
+        }
+        result += core::ErrorColors::format("or `{}`", validLayers.back().shortName(gs));
+        return result;
+    }
```

Comment:
Minor, not blocking: this doesn't use any members of the enclosing class—it could be a top-level helper function instead of a private method if you wanted it to be.